### ISSUE TITLE
UnsupportOperationException Thrown When using sd.loss.ctcLoss

### ISSUE THREAD

**Issue by AllenWGX** [deeplearning4j/deeplearning4j#9826]
#### Issue Description
I wrote a demo to test sd.loss.ctcLoss, and got an UnsupportOperationException when getting the output. Pls help to check the code.

```
public class TestCtc {
	private SameDiff sd = null;
	
	@Before
	public void init() {
		sd = SameDiff.create();
	}
	
	@Test
	public void test() {
		INDArray labelsND = Nd4j.createFromArray(new int[][] {{1917, 3468, 1024, 2744, 4092, 2613,  112,  922, 4785, 1675},
			{119, 16, 202, 2352, 2945, 3468, 2744, 112, 0, 0}});
		INDArray labels_len_ND =  Nd4j.createFromArray(new int[] {10, 8});
		INDArray logits_len_ND =  Nd4j.createFromArray(new int[] {155, 155});
		SDVariable labels = sd.constant(labelsND);
		SDVariable logits = sd.random.normal(0, 1, DataType.FLOAT, new long[] {2, 155, 5530});
		SDVariable labels_len = sd.constant(labels_len_ND);
		SDVariable logits_len = sd.constant(logits_len_ND);
		SDVariable ctc = sd.loss.ctcLoss("ctcLoss", labels, logits, labels_len, logits_len);
		//
		System.out.println(ctc.eval());
	}
	
	@After
	public void destroy() {
		
	}
}
```
![ctc](https://user-images.githubusercontent.com/23058056/198360103-38fb06a0-0e6e-4520-a328-96588f746ba8.png)

**BTW:** I wonder if ctcLoss in SameDiff supports training, or only support forward pass ? In addition, after checking the ctc code in libnd4j, I find that ctc_beam_search_decoder has already been implemented but it seems that SameDiff does not have the corresponding op ?

#### Version Information

Please indicate relevant versions, including, if relevant:

* Deeplearning4j version: 1.0.0-SNAPSHOT
* Platform information (OS, etc): Win10
* CUDA version, if used: Not Used
* NVIDIA driver version, if in use: Not Used

#### Additional Information

Where applicable, please also provide:
* pom.xml file or similar (also in a Gist): [pom.xml](https://gist.github.com/AllenWGX/0577bd01a0cded0e1faff3c57969b374)
